### PR TITLE
Fix: testing documentation

### PR DESCRIPTION
**Pull request by Olavhaasie**
## Description
I have removed the tutorials for writing tests, since these tutorials already exist by ROS itself, were not specific for March and they were horribly outdated and wrong. I replaced them with external links on how to write tests for Python, C++ and ROS nodes. I also added some rules on structuring tests.

## Changes
* Removed outdated testing tutorials
* Added external links on how to test
* Added rules on structuring tests

### CODE REVIEW

File: doc/development/testing.rst [hunk 1]
```diff
@@ -59,14 +59,25 @@ General guidelines to writing tests
...
+#. Make a test for every edge case!
+#. Usually, your test code is about as long as your normal code.
+
+Next to these rules there exist some conventions in structuring tests:
+
+#. Put your tests in a separate directory next to ``src/`` named ``test/``.
+#. The same as for all filenames, name the test files using snake_case.
+#. Put rostest ``.test`` launch files inside ``test/launch/``.
+#. Put rostest source files inside ``test/rostest/``.
+#. If writing unittests and rostests for Python, put the unit tests inside ``test/nosetests/``.
```
Comment (RutgerVanBeek) [inline doc/development/testing.rst]:
```suggestion
#. If writing unittests and rostests for Python, put the unit tests inside ``test/unittests/``.
```
I guess this is what you mean.

**Review by RutgerVanBeek — CHANGES_REQUESTED** [review]
One small comment, furthermore nice.

File: doc/development/testing.rst (hunk 1 above)
Comment (Olavhaasie) [reply doc/development/testing.rst]: Yes that would be nice, I looked at the `march_gait_selection` package which had `nosetests`, so we should maybe change the name of that directory.

**Review by RutgerVanBeek — APPROVED** [review]
